operator/controllers/resources/role_controller.go

```diff
@@ -75,13 +78,35 @@ func (r *RoleReconciler) Upsert(ctx context.Context, obj kclient.Object) error {
 	}
 	teleportResource := k8sResource.ToTeleport()
 
-	_, err := r.TeleportClient.GetRole(ctx, teleportResource.GetName())
+	existingResource, err := r.TeleportClient.GetRole(ctx, teleportResource.GetName())
 	if err != nil && !trace.IsNotFound(err) {
 		return trace.Wrap(err)
 	}
-	if trace.IsNotFound(err) && !hasOriginLabel(obj) {
-		return addOriginLabelToK8SObject(ctx, r.Client, obj)
+
+	updateCondition := func(condition metav1.Condition) error {
+		meta.SetStatusCondition(&k8sResource.Status.Conditions, condition)
+		err := r.Status().Update(ctx, k8sResource)
+		if err != nil {
+			return trace.Wrap(err)
+		}
+		return nil
+	}
+
+	err = checkOwnership(existingResource, updateCondition)
+	if err != nil {
+		return trace.Wrap(err)
```

Comment:
:nail_care: We can use the comma ok pattern here
```suggestion
	if err := checkOwnership(existingResource, updateCondition); err != nil {
		return trace.Wrap(err)
```